prototype/utils/serial_port.py

```diff
@@ -0,0 +1,100 @@
+import serial
+import time
+import struct
+
+
+'''
+Custom serial class to write and read data
+'''
+class CustomSerial:
+    """Custom serial class for communication
+    """
+
+    def __init__(self, address: str, baudRate: int, openNow: bool = True):
```

Comment:
The parameters address and baudrate are only used when openNow is true, should we make those parameters optionals?